src/builtins/sci/index.ts

```diff
@@ -7,176 +7,75 @@ import {validate, allDefined} from '../../util/validations';
+    const errorMessage =
+      '`functional-unit` should be provided in your global config';
 
-    return inputs.map(input => {
-      const safeInput = validateInput(input);
-      const inputWithConfigs = Object.assign(
-        {},
-        input,
-        safeInput,
-        validatedConfigs
-      );
+    const schema = z
+      .object({
+        'functional-unit': z.string(),
+      })
+      .refine(data => data['functional-unit'], {
+        message: errorMessage,
+      });
 
-      return {
-        ...input,
-        ...tuneInput(inputWithConfigs),
-      };
-    });
+    return validate<z.infer<typeof schema>>(schema, config);
   };
 
   /**
-   * Given an input, tunes it and returns the tuned input.
+   * Calculate the total emissions for a list of inputs.
    */
-  const tuneInput = (input: PluginParams) => {
-    const sciPerSecond = calculateSciSeconds(input);
-    const factor = getFunctionalUnitConversionFactor(input);
-
-    if (!input['functional-unit-time']) {
+  const execute = (inputs: PluginParams[]): PluginParams[] => {
+    return inputs.map(input => {
+      const safeInput = validateInput(input);
+      const sci =
+        safeInput['carbon'] > 0
+          ? safeInput['carbon'] / input[globalConfig['functional-unit']]
+          : 0;
       return {
-        carbon: input['carbon'] ?? sciPerSecond,
-        sci: sciPerSecond / factor,
+        ...input,
+        sci,
       };
-    }
-
-    const functionalUnitTime = parseTime(input);
-    const sciTimed = convertSciToTimeUnit(sciPerSecond, functionalUnitTime);
-    const sciTimedDuration = sciTimed * functionalUnitTime.value;
-
-    return {
-      carbon: input['carbon'] ?? sciPerSecond,
-      sci: sciTimedDuration / factor,
-    };
+    });
   };
 
   /**
-   * Gets the conversion factor based on the functional unit specified in the input.
-   * If the 'functional-unit' exists in the input and is not 'none' or an empty string,
-   * returns the value; otherwise, defaults to 1.
+   * Checks for fields in input.
    */
-  const getFunctionalUnitConversionFactor = (input: PluginParams): number => {
-    const functionalUnit = input['functional-unit'];
-
-    return functionalUnit in input &&
-      input[functionalUnit] !== 'none' &&
-      input[functionalUnit] !== ''
-      ? input[functionalUnit]
-      : 1;
-  };
+  const validateInput = (input: PluginParams) => {
+    const message = `'carbon' and ${globalConfig['functional-unit']} should be present in your input data.`;
 
-  /**
-   * Converts the given sci value from seconds to the specified time unit.
-   */
-  const convertSciToTimeUnit = (
-    sciPerSecond: number,
-    functionalUnitTime: {unit: string; value: number}
-  ): number => {
-    const conversionFactor = TIME_UNITS_IN_SECONDS[functionalUnitTime.unit];
+    const validatedConfig = validateConfig(globalConfig);
 
-    if (!conversionFactor) {
+    if (!(validatedConfig['functional-unit'] in input)) {
```

Comment:
please, also check if the value of `input[validatedConfig['functional-unit']]` is > 0

Reply:
done